Tests/mock_server.py

```diff
@@ -362,101 +371,159 @@ def start(self, playbook_id, path=None, record=False, thread_index=0, prints_man
+        Starts mitmdump service on the remote service
+        """
+        self.ami.call(['sudo', 'systemctl', 'start', 'mitmdump'])
 
-        # Create mock files directory
-        silence_output(self.ami.call, ['mkdir', os.path.join(path, get_folder_path(playbook_id))], stderr='null')
+    def get_mitmdump_service_status(self, prints_manager, thread_index: int) -> None:
+        """
+        Safely extract the current mitmdump status and logs it
+        Args:
+            prints_manager(ParallelPrintsManager): Prints manager to synchronize parallel prints.
+            thread_index (int): Index of the relevant thread, to make printing readable.
+        """
+        try:
+            output = self.ami.check_output('systemctl status mitmdump'.split(), stderr=STDOUT)
+            prints_manager.add_print_job(f'mitmdump service status output:\n{output.decode()}', print_error,
+                                         thread_index)
+        except CalledProcessError as exc:
+            prints_manager.add_print_job(f'mitmdump service status output:\n{exc.output.decode()}', print_error,
+                                         thread_index)
+
+    def prepare_proxy_start(self,
+                            path: str,
+                            playbook_id: str,
+                            record: bool,
+                            prints_manager,
+                            thread_index: int) -> bool:
+        """
+        Does all needed preparation for starting the proxy service which include:
+        - Creating new tmp directory on remote machine if in record mode and moves the problematic keys file in to it
+        - Creating the mitmdump_rc file that includes the script mode, keys file path, mock file path and log file path
+          for the mitmdump service and puts it in '/home/ec2-user/mitmdump_rc' in the remote machine.
+        - starts the systemd mitmdump service
 
-        repo_problem_keys_filepath = os.path.join(
-            self.repo_folder, get_folder_path(playbook_id), 'problematic_keys.json'
-        )
-        current_problem_keys_filepath = os.path.join(path, get_folder_path(playbook_id), 'problematic_keys.json')
+        Args:
+            path: the path to the temp folder in which the record files should be created
+            playbook_id: The ID of the playbook that is tested
+            record: Indicates whether this is a record run or not
+            prints_manager: A prints manager instance
+            thread_index: The index of the current thread
+        """
+        path = path or self.current_folder
+        folder_path = get_folder_path(playbook_id)
+
+        repo_problem_keys_path = os.path.join(AMIConnection.REMOTE_HOME,
+                                              self.repo_folder,
+                                              folder_path,
+                                              'problematic_keys.json')
+        current_problem_keys_path = os.path.join(AMIConnection.REMOTE_HOME,
```

Comment:
During `record` mode, the path would be something like `/tmp/Mocks/<playbook-id>/problematic_keys.json` so I don't think having the path start with `AMIConnection.REMOTE_HOME` makes sense here...